src/libutil/util.cc

```diff
@@ -364,10 +364,13 @@ void writeFile(const Path & path, std::string_view s, mode_t mode)
         e.addTrace({}, "writing file '%1%'", path);
         throw;
     }
+    if (sync)
+        fd.fsync();
+    fd.close();
```

Comment:
I thinks that's worth a comment explaining why it's explicitly closed

Reply:
Added an extra comment here.